pkg/cli/completion.go

```diff
@@ -85,6 +86,98 @@ func (a *completion) complete(cmd *cobra.Command, args []string, toComplete stri
 	return removeExistingArgs(result, args), a.successDirective
 }
 
+// checkProjectPrefix
+// called on a completion object, which already has the completion function defined
+// this function modifies that completion function, wraps it with some logic to handle projects
+func (a *completion) checkProjectPrefix() *completion {
```

Comment:
Yes, this can be refactored into the .complete(). There are maybe 2 or so completions where we do not want project completion however. Can make a condition to disable it, and have it enabled by default. Great idea, thanks!